Snowflake.Data/Core/FileTransfer/StorageClient/SFS3Client.cs

```diff
@@ -562,5 +536,47 @@ private SFFileMetadata HandleFileHeaderErr(Exception ex, SFFileMetadata fileMeta
             }
             return fileMetadata;
         }
+
+        /// <summary>
+        /// Handle file upload error.
+        /// </summary>
+        /// <param name="ex">Exception from file header.</param>
+        /// <param name="fileMetadata">The file metadata.</param>
+        /// <returns>The file metadata.</returns>
+        private SFFileMetadata HandleUploadFileErr(Exception ex, SFFileMetadata fileMetadata)
+        {
+            AmazonS3Exception err = (AmazonS3Exception)ex;
```

Comment:
Let's log the error here as it is reused in many places and can be helpful when dealing with some issues on the customer side

Reply:
Done, I also logged it for the other error handling functions